vision.md

```diff
@@ -0,0 +1,407 @@
+    - This does not preclude other forms of extensions in other languages (i.e.
+      a bridge over MQTT to python)
+1. Starting a thin-edge.io binary for development use or production use is not
+   interactive
+    - Starting a thin-edge.io binary for setup or similar purposes may be
+      interactive
+1. Components can persist data using the framework
+    - to persist data between restarts of the deployment
+    - to cache data during network outages
+    - to provide operation checkpoints during sensitive operations
+1. User-configurable logging is provided via the framework
+    - A user is able to configure logging per-component as well as globally for
+      the framework
+
+### Capabilities
+
+<!--
+The following list of capabilities should be considered non-final. It might be
+extended in the future.
+-->
+
+In combination with IoT platforms, thin-edge.io is a foundation for enabling
+devices with the following capabilities:
+
+1. Connectivity to the major cloud providers in the IIOT space
+    - for effortless and secure edge device lifecycle management for single
+      and device fleets
+    - for low-touch provisioning of thin edge devices
+    - for local and remote configuration
+    - for local and remote maintenance including remote access
+      (monitoring/troubleshooting)
+    - decommissioning of thin devices (e.g. for security compromised or end of
+      life devices)
+1. "South bound" connectivity to devices via the major protocols in the IIOT industry
+1. On-device data preprocessing
+    1. Analytical (timeseries analysis, ML, etc...)
+    2. Mathematical operations (avg, sum, etc...)
+    3. Logical operations
+1. On-device data generation (e.g. generating of events)
+1. Device management functionality
+    1. OS updates
+    1. Package updates
+    1. Configuration updates (Of both the thin-edge.io deployment as well as other software)
+    1. Firmware updates for "south bound"/child devices
+1. Device interaction
+    1. File upload/download
+    1. Filesystem listings
+    1. General system informations
+1. Persist data as given by components in different databases (e.g. PostgreSQL, Sqlite, MongoDB, Redis, Memcached, ...)
+1. Documentation of each component and their configuration
+    - This includes information on how to configure each configurable aspect of the component and its valid states
+    - This also includes all message types that the framework knows about
+    - Users with custom thin-edge.io deployments must be able to generate such a documentation themselves
+
+## Closing words
+
+thin-edge.io offers a unique approach to unify the needs of both the IT and OT
+world by offering a platform design focused on efficiency robustness and
+security while offering the highest level of extensibility.
+
+- We are not restricting users towards one specific software artifact type,
+  package manager, programming language or message payload to be used on the
+  device
```

Comment:
Should we add cloud platform here as well?

Reply:
We need to be consistent, if we are more comfortable calling thin-edge.io as "framework" rather than a "platfomr"- then we should use the same throughout the document.